src/corehost/cli/hostpolicy.cpp

```diff
@@ -273,16 +308,22 @@ int run(const arguments_t& args)
 
     coreclr::unload();
 
-    // Finish breadcrumb writing
-    writer.end_write();
+    if (breadcrumbs_enabled)
+    {
+        // Finish breadcrumb writing
+        writer.end_write();
+    }
 
     return exit_code;
 }
 
 SHARED_API int corehost_load(host_interface_t* init)
 {
     trace::setup();
-    
+
+    // Re-initialize global state in case of re-entry
```

Comment:
Do we have a re-entry test case?

Reply:
Yes, the unit test that was added calls back in a second time because its buffer was too small the first time